src/libexpr/eval.cc

```diff
@@ -1508,6 +1508,10 @@ void ExprLambda::eval(EvalState & state, Env & env, Value & v)
 
 void EvalState::callFunction(Value & fun, size_t nrArgs, Value * * args, Value & vRes, const PosIdx pos)
 {
+    if (depth > 10000)
+        error("stack depth exhausted").atPos(pos).template debugThrow<EvalError>();
```

Comment:
Is this message OK? Maybe it should say "potential infinite recursion" or just "stack overflow".

Reply:
I made it say "stack overflow" instead.